### PR TITLE
refactor: move package to build

### PR DESCRIPTION
**Pull request by nliu-ms**
build/appPackage/appPackage.{env}.zip
build/appPackage/manifest.{env}.zip

Trigger from:
- Remote: zip package / provision / publish
- Local: local debug

### CODE REVIEW

**Comment by nliu-ms** [comment]
E2E TEST: https://github.com/OfficeDev/TeamsFx/blob/149cdd67807a59f3325ba5e6e167d0e9897df727/packages/cli/tests/e2e/multienv/TestRemoteHappyPath.tests.ts#L132

File: packages/fx-core/src/plugins/resource/appstudio/plugin.ts [hunk 1]
```diff
@@ -590,26 +591,35 @@ export class AppStudioPluginImpl {
...
     if (!ctx.envInfo?.envName) {
       throw new Error("Failed to get target environment name from plugin context.");
     }
 
     const appDirectory = await getAppDirectory(ctx.root);
-    const zipFileName: string = isMultiEnvEnabled()
-      ? `${ctx.root}/${AppPackageFolderName}/appPackage.${ctx.envInfo.envName}.zip`
-      : `${ctx.root}/${AppPackageFolderName}/appPackage.zip`;
+    let zipFileName: string;
+    if (isMultiEnvEnabled()) {
```
Comment (HuihuiWu-Microsoft) [inline packages/fx-core/src/plugins/resource/appstudio/plugin.ts:603]: So this partially applies to multi-env, app package will be generated when publish/zip/provision, but the manifest file will not be generated?

Comment (nliu-ms) [reply packages/fx-core/src/plugins/resource/appstudio/plugin.ts:603]: It applies to multi-env, the manifest file will also be generated from line 684

Comment (HuihuiWu-Microsoft) [reply packages/fx-core/src/plugins/resource/appstudio/plugin.ts:603]: Sorry, typo. I mean non multi env. If it's not multi-env,  the app package will still be generated when publish/zip/provison in current logic, right?

Comment (nliu-ms) [reply packages/fx-core/src/plugins/resource/appstudio/plugin.ts:603]: no, for non multi-env, only zip/publish will trigger build. Provision/local debug won't.

**Review by HuihuiWu-Microsoft — DISMISSED** [review]

**Comment by nliu-ms** [comment]
E2E passed: https://github.com/OfficeDev/TeamsFx/actions/runs/1433926695

**Review by jayzhang — APPROVED** [review]

**Review by a1exwang — APPROVED** [review]

**Pull request merged**